lib/wmi-lite/wmi.rb

```diff
@@ -61,12 +70,14 @@ def query_with_context(wql_query, diagnostic_class_name = nil)
 
     def start_query(wql_query, diagnostic_class_name = nil)
       result = nil
-      connect_to_namespace
-      begin
-        result = @connection.ExecQuery(wql_query)
-        raise_if_failed(result)
-      rescue WIN32OLERuntimeError => native_exception
-        raise WmiException.new(native_exception, :ExecQuery, @namespace, wql_query, diagnostic_class_name)
+      Timeout::timeout(@timeout, WmiTimeoutException) do
```

Comment:
So to clarify, Timeout creates a new thread, and then kills it if things take too long. We should definitely add some comments to clarify that so everyone can understand the impact on COM / OS thread state.